jose/encrypt.go

```diff
@@ -19,12 +19,30 @@ type PasswordPrompter func(s string) ([]byte, error)
 // the parse of the key will fail.
 var PromptPassword PasswordPrompter
 
-// EncryptJWK returns the given JWK encrypted with the default encryption
+// Encrypt returns the given data encrypted with the default encryption
 // algorithm (PBES2-HS256+A128KW).
-func EncryptJWK(jwk *JSONWebKey, passphrase []byte) (*JSONWebEncryption, error) {
-	b, err := json.Marshal(jwk)
+func Encrypt(data []byte, opts ...Option) (*JSONWebEncryption, error) {
+	ctx, err := new(context).apply(opts...)
 	if err != nil {
-		return nil, errors.Wrap(err, "error marshaling JWK")
+		return nil, err
+	}
+
+	var passphrase []byte
+	switch {
+	case len(ctx.password) > 0:
+		passphrase = ctx.password
+	case ctx.passwordPrompter != nil || PromptPassword != nil:
```

Comment:
This would be simpler with two separate cases

Reply:
I was fixing a linter error, which was the direct fix. 
Fixed with c855ec6